File: solutions/typescript/collatz-conjecture/1/collatz-conjecture.ts
export function steps(count: number): number {
  if (count <= 0 || !Number.isInteger(count)) {
    throw new Error('Only positive integers are allowed');
  }
  
  if (count === 1) {
    return 0;
  }
  
  let stepsCount = 0;
  let n = count;
  
  while (n > 1) {
    n = n % 2 === 0 ? n / 2 : n * 3 + 1;
    stepsCount++;
  }
  
  return stepsCount;
}

File: solutions/vbnet/collatz-conjecture/1/CollatzConjecture.vb
Public Module CollatzConjecture
    Public Function Steps(ByVal number As Integer) As Integer
        If number < 1 Then
            Throw New ArgumentOutOfRangeException(NameOf(number), "Only positive integers are allowed")
        End If

        Dim current As Integer = number
        Dim stepCount As Integer = 0

        While current <> 1
            If current Mod 2 = 0 Then
                current = current \ 2
            Else
                current = current * 3 + 1
            End If
            stepCount += 1
        End While

        Return stepCount
    End Function
End Module